src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,208 @@
+    }
+
+    @Test
+    void loginIsNull_NotOk() {
+        User user = new User();
+        user.setPassword("password");
+        user.setAge(33);
+
+        assertThrows(CustomException.class, () -> registrationService.register(user),
+                "Login can't be null, login must be at list 6 characters long");
+        assertTrue(Storage.people.isEmpty(), "Storage must be empty");
+    }
+
+    @Test
+    void passwordIsNull_NotOk() {
+        User user = new User();
+        user.setLogin("testlogin");
+        user.setAge(33);
+
+        assertThrows(CustomException.class, () -> registrationService.register(user),
+                "Password is empty, password must be at list 6 characters long");
+        assertTrue(Storage.people.isEmpty(), "Storage must be empty");
+    }
+
+    @Test
+    void passwordIsValid_Ok() {
+        User user = new User();
+        user.setLogin("testlogin");
+        user.setPassword("password");
+        user.setAge(33);
+        registrationService.register(user);
+
+        assertTrue(user.getLogin().length() >= MIN_PASSWORD_LENGTH,
+                "Password must be at list 6 characters long");
+        User result = storageDao.get(user.getLogin());
+        assertEquals(user, result, "User must be added to storage");
+    }
+
+    @Test
+    void passwordIsShort_NotOk() {
+        User user = new User();
+        user.setLogin("testlogin");
+        user.setPassword("pass");
+        user.setAge(33);
+
+        assertThrows(CustomException.class, () -> registrationService.register(user),
+                "Password must be at list 6 characters long");
+        assertTrue(Storage.people.isEmpty(), "Storage must be empty");
+    }
+
+    @Test
+    void passwordIsLong_Ok() {
+        User user = new User();
+        user.setLogin("testlogin");
+        user.setPassword("passwordislong");
+        user.setAge(33);
+
+        assertTrue(user.getLogin().length() >= MIN_PASSWORD_LENGTH,
+                "Password must be at list 6 characters long");
+        registrationService.register(user);
+        assertNotNull(storageDao.get(user.getLogin()), "User must be added to storage");
+    }
+
+    @Test
+    void ageIsNull_NotOk() {
+        User user = new User();
+        user.setLogin("testlogin");
+        user.setPassword("password");
+
+        assertThrows(CustomException.class, () -> registrationService.register(user),
+                "User age is null, must be minimum 18");
+        assertTrue(Storage.people.isEmpty(), "Storage must be empty");
+    }
+
+    @Test
+    void ageIsUnderRequirements_NotOk() {
+        User user = new User();
+        user.setLogin("testlogin");
+        user.setPassword("password");
+        user.setAge(17);
+
+        assertThrows(CustomException.class, () -> registrationService.register(user),
+                "User age must be at list 18 y.o.");
+        assertTrue(Storage.people.isEmpty(), "Storage must be empty");
+    }
+
+    @Test
+    void ageIsHigh_Ok() {
```

Comment:
<img width="762" alt="image" src="https://github.com/mate-academy/jv-registration-validation/assets/81781479/b9e76566-4bdc-4220-b28e-2c0a302c1121">

Reply:
not fixed. Where are method names